internals/api/service.go

```diff
@@ -23,13 +23,32 @@ var (
 	)
 )
 
+type ServiceType string
+
+// Service types
+const (
+	ServiceTypeAWS ServiceType = "aws"
+	ServiceTypeRSA ServiceType = "rsa"
+)
+
+type ServiceMetaDataKey string
+
+// Service metadata keys
+const (
+	ServiceMetaDataAWSKMSKey ServiceMetaDataKey = "kms_key"
+	ServiceMetaDataAWSRole   ServiceMetaDataKey = "role"
+)
+
 // Service represents a service account on SecretHub.
 type Service struct {
-	AccountID   *uuid.UUID `json:"account_id"`
-	ServiceID   string     `json:"service_id"`
-	Repo        *Repo      `json:"repo"`
-	Description string     `json:"description"`
-	CreatedBy   *uuid.UUID `json:"created_by,omitempty"`
+	AccountID   *uuid.UUID        `json:"account_id"`
+	ServiceID   string            `json:"service_id"`
+	Repo        *Repo             `json:"repo"`
+	Description string            `json:"description"`
+	CreatedBy   *uuid.UUID        `json:"created_by,omitempty"`
+	CreatedAt   time.Time         `json:"created_at"`
+	Type        ServiceType       `json:"type"`
+	MetaData    map[string]string `json:"meta_data,omitempty"`
```

Comment:
In my head it's `metadata` without a space instead of `meta data`. It also seems to be what other APIs use as name, e.g. Stripe API.